cpp/models/abm/location.h

```diff
@@ -97,20 +97,46 @@ class Location
 {
 public:
     /**
-     * @brief Construct a Location of a certain type.
-     * @param type The #LocationType.
-     * @param index The index of the Location.
-     * @param num_agegroups The number of age groups in the model
-     * @param num_cells [Default: 1] The number of Cell%s in which the Location is divided.
+     * @brief Construct a Location of a certain LocationId.
+     * @param[in] loc_id The #LocationId.
+     * @param[in] num_agegroups [Default: 1] The number of age groups in the model.
+     * @param[in] num_cells [Default: 1] The number of Cell%s in which the Location is divided.
      */
-    Location(LocationId loc_id, size_t num_agegroups, uint32_t num_cells = 1);
+    Location(LocationId loc_id, size_t num_agegroups = 1, uint32_t num_cells = 1);
 
-    Location(LocationType loc_type, uint32_t loc_index, size_t num_agegroups, uint32_t num_cells = 1)
+    /**
+     * @brief Construct a Location with provided parameters. 
+     * @param[in] loc_type The #LocationType.
+     * @param[in] index The index of the Location.
+     * @param[in] num_agegroups [Default: 1] The number of age groups in the model.
+     * @param[in] num_cells [Default: 1] The number of Cell%s in which the Location is divided.
+     */
+    Location(LocationType loc_type, uint32_t loc_index, size_t num_agegroups = 1, uint32_t num_cells = 1)
         : Location(LocationId{loc_index, loc_type}, num_agegroups, num_cells)
     {
     }
 
-    Location copy_location();
+    /**
+     * @brief Return a copy of the current Location object.
+     * @param[in] other The original #Location.
+     */
+    Location(const Location& other)
```

Comment:
Yes. This is due to the usage of the new mutex variable `m_mut` which deletes the default copy constructor.

Reply:
Ah, yes, I didn't see that before. Perhaps we should think about out-sourcing the mutex and the vector of persons to an extra struct, so that we can avoid writing any SMF for the Location? Also, the vector of Persons should probably be locked for the copy-duration, although this is not really a use-case for use, but just to make sure. @dabele Do you have ideas for this?

Reply:
Ultimately, the solution is to not store the persons and avoid the mutex. But that's a different discussion we need to have at some point.

I agree you would probably want the mutex in a wrapper so you only need to write explicit copy constructor for one thing instead of all the data members of Location. Easy to forget something. 

If you want to lock the mutex while copying, it needs to be declared mutable. I don't really care, since copying locations or anything else while the simulation is running is completely invalid and I don't see why someone would do it except accidentally, in which case a crash is a good thing. 

(In fact I kind of liked non copyable Locations since I detected a number of accidental copies that cost a lot of performance. From a safety point of view, I would prefer locations and persons not to have a copy constructor, but some explicit copy function, but I see how that would be too difficult to maintain without the help of default copy constructors. I think all we can do is watch out for accidental copies like `auto loc = person.get_location()`)

Reply:
Yes, I agree. Since it's not really in the scope of this issue. I would probably refer this development to issue #757. I also added your comments there for reference.